frontend/packages/console-dynamic-plugin-sdk/src/app/__tests__/AppInitSDK.spec.tsx

```diff
@@ -1,113 +1,90 @@
+    useReduxStoreMock.mockReturnValue(storeContext);
+    return render(
+      <AppInitSDK configurations={config}>
+        <ChildComponent />
+      </AppInitSDK>,
+    );
   };
 
-  let useReduxStoreSpy;
-  let configSetupSpy;
-  beforeEach(() => {
-    useReduxStoreSpy = jest.spyOn(hooks, 'useReduxStore');
-    configSetupSpy = jest.spyOn(configSetup, 'setUtilsConfig');
-  });
-
   afterEach(() => {
-    jest.resetAllMocks();
+    jest.clearAllMocks();
   });
 
   it('should not wrap children with Provider', () => {
-    useReduxStoreSpy.mockImplementation(() => ({ store, storeContextPresent: true }));
-    const wrapper = shallow(
-      <AppInitSDK configurations={mockConfig}>
-        <div data-test-id="child-id">Hello!!</div>
-      </AppInitSDK>,
-    );
-    expect(wrapper.find(Provider)).toHaveLength(0);
-    expect(wrapper.find('[data-test-id="child-id"]')).toHaveLength(1);
+    renderAppInitSDK(mockConfig, { store, storeContextPresent: true });
+
+    expect(mockProvider).not.toHaveBeenCalled();
+    expect(screen.getByText('Hello, OpenShift!')).toBeVisible();
   });
 
-  it('should wrap children with Provider', () => {
-    useReduxStoreSpy.mockImplementation(() => ({ store, storeContextPresent: false }));
-    const wrapper = shallow(
-      <AppInitSDK configurations={mockConfig}>
-        <div data-test-id="child-id">Hello!!</div>
-      </AppInitSDK>,
-    );
-    expect(wrapper.find(Provider)).toHaveLength(1);
-    expect(wrapper.find('[data-test-id="child-id"]')).toHaveLength(1);
+  it('should wrap children with a Provider if no store context is present', () => {
+    renderAppInitSDK(mockConfig, { store, storeContextPresent: false });
+
+    expect(mockProvider).toHaveBeenCalled();
+    expect(screen.getByText('Hello, OpenShift!')).toBeVisible();
   });
 
-  it('should call the hook useReduxStore', () => {
-    useReduxStoreSpy.mockImplementation(() => ({ store, storeContextPresent: true }));
-    shallow(
-      <AppInitSDK configurations={mockConfig}>
-        <div data-test-id="child-id">Hello!!</div>
-      </AppInitSDK>,
-    );
-    expect(useReduxStoreSpy).toHaveBeenCalled();
-    expect(useReduxStoreSpy).toHaveBeenCalledTimes(1);
+  it('should call the useReduxStore hook', () => {
+    renderAppInitSDK(mockConfig);
+
+    expect(useReduxStoreMock).toHaveBeenCalledTimes(1);
   });
 
-  it('should call the util setUtilsConfig with proper config', () => {
-    useReduxStoreSpy.mockImplementation(() => ({ store, storeContextPresent: true }));
-    mount(
-      <AppInitSDK configurations={mockConfig}>
-        <div data-test-id="child-id">Hello!!</div>
-      </AppInitSDK>,
-    );
-    expect(configSetupSpy).toHaveBeenCalled();
-    expect(configSetupSpy).toHaveBeenCalledTimes(1);
-    expect(configSetupSpy).toHaveBeenCalledWith({ appFetch: mockConfig.appFetch });
+  it('should call the setUtilsConfig utility with the proper config', () => {
+    renderAppInitSDK(mockConfig);
```

Comment:
I think it's better to consistently explicitly define this config 

```suggestion
    renderAppInitSDK(mockConfig, { store, storeContextPresent: true });
```